code/modules/mining/lavaland/loot/tendril_loot.dm

```diff
@@ -174,9 +176,20 @@
 	if(istype(next, /turf/simulated/floor/plating/lava/smooth) || istype(current, /turf/simulated/floor/plating/lava/smooth)) //We can move from land to lava, or lava to land, but not from land to land
 		..()
 	else
-		to_chat(user, "<span class='warning'>Boats don't go on land!</span>")
+		if(last_message_time + 1 SECONDS < world.time)
+			to_chat(user, "<span class='warning'>Boats don't go on land!</span>")
 		return FALSE
 
+/obj/vehicle/lavaboat/user_buckle_mob(mob/living/M, mob/user)
+	M.weather_immunities += "lava"
+	. = ..()
+	// If you're buckled, you won't be subject to lava
+
+
+/obj/vehicle/lavaboat/unbuckle_mob(mob/living/buckled_mob, force)
```

Comment:
Should `M` and `buckled_mob` be the same variable name for readability?

Reply:
These were the variables from the super calls, honestly I think vehicles need a refactor before anything else xd